app/assets/javascripts/discourse/app/initializers/opengraph-tag-updater.js

```diff
@@ -20,7 +20,8 @@ export default {
     this.ogUrl.setAttribute("content", getAbsoluteURL(data.url));
   },
 
-  teardown() {
-    this.appEvents.off("page:changed", this, this.updateOgAttributes);
+  teardown(container) {
+    const appEvents = container.lookup("service:app-events");
+    appEvents.off("page:changed", this, this.updateOgAttributes);
```

Comment:
valid for both files, could you check this is actually the same appEvents in initialise in and in teardown I have a feeling this is not the case and that as such `off` does nothing, it was on my list to check.